scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -241,6 +241,255 @@ fun Bitmap.zoom(
     return this.fragment(fromSegments, toSegments, toSize)
 }
 
+fun Bitmap.zoomOffset(
+    centerSize: Size,
+    previewSize: Size,
+    cardFinder: Rect,
+    toCenterDimension: Int,
+    toBorderWidth: Int
+): Bitmap {
+    require(cardFinder.left > 0 && cardFinder.right < previewSize.width && cardFinder.top > 0 && cardFinder.bottom < previewSize.height) {
+        "Card viewfinder window extends beyond preview boundary"
+    }
+    val newLeft = cardFinder.left * this.width / previewSize.height
+    val newTop = cardFinder.top * this.height / previewSize.height
+    val newRight = cardFinder.right * this.width / previewSize.width
+    val newBottom = cardFinder.bottom * this.height / previewSize.height
+
+    val scaledCardFinderRect = Rect(newLeft, newTop, newRight, newBottom)
```

Comment:
let's keep these all together using a rect.
```suggestion
    val scaledPreviewSize = previewSize.scaleAndCenterWithin(this.size())
    val previewScale = scaledPreviewSize.width().toFloat() / previewSize.width
    val scaledCardFinderRect = Rect(
        (cardFinder.left * previewScale).roundToInt(),
        (cardFinder.top * previewScale).roundToInt(),
        (cardFinder.right * previewScale).roundToInt(),
        (cardFinder.bottom * previewScale).roundToInt()
    )
```